### ISSUE TITLE
Is there a way to emit only the kernel ast on the device side

### ISSUE THREAD

**Issue by malixian** [intel/llvm#4880]
I have a dpc++ code and want to emit only the device code AST. I use a clang command follows:

> clang++ -Xclang  -ast-dump -fsycl  vector_add.cpp

When executing this command,  there will be a lot of AST host code that I don’t need. 
So is there any way to generate only the kernel ast?

**Comment by AlexeySachkov** [comment]
Tagging @Fznamznon to correct me if I'm wrong, but it looks like `-fsycl-device-only` is an argument you are looking for, @malixian

**Comment by Fznamznon** [comment]
I don't think we have a way to do that. Even if `-fsycl-device-only` is passed - AST will be built and dumped by `ast-dump` for the whole translation including host code.

**Comment by malixian** [comment]
> I don't think we have a way to do that. Even if `-fsycl-device-only` is passed - AST will be built and dumped by `ast-dump` for the whole translation unit including host code.

Yes, you are right, I have tried it.
